rca/research/models.py

```diff
@@ -282,6 +282,18 @@ def get_research_news(self):
         for value in self.research_news.select_related("page"):
             if value.page.live:
                 page = value.page.specific
+                page_type = None
+                if page.__class__.__name__ == "GuidePage":
+                    page_type = "GUIDE"
+                if page.__class__.__name__ == "ProjectPage":
+                    page_type = "PROJECT "
+                if page.__class__.__name__ == "ResearchCentrePage":
+                    page_type = "RESEARCH CENTRE"
+                if page.__class__.__name__ == "ShortCoursePage":
+                    page_type = "SHORT COURSE"
+                if page.__class__.__name__ == "ProgrammePage":
+                    page_type = "PROGRAMME"
+
```

Comment:
Personally I think a mapping dict makes for neater code:

```python
page_type = None
page_type_mapping = {
    "GuidePage": "GUIDE",
    "ProjectPage": "PROJECT",
    "ResearchCentrePage": "RESEARCH CENTRE",
    "ShortCoursePage": "SHORT COURSE",
    "ProgrammePage": "PROGRAMME",
}
if page.__class__.__name__ in page_type_mapping:
    page_type = page_type_mapping[page.__class__.__name__]
```

Reply:
Yea much better, thanks